### PR TITLE
BAM to fastq conversion

### PR DESCRIPTION
**Pull request by skanwal**
Added tool definition for biobambam2-bamtofastq.

### CODE REVIEW

File: tools/bamtofastq/2.0.0/bamtofastq__2.0.0.cwl [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+  level:
+    label: level
+    type: int?
+    doc: "compression setting if gz=1 (-1=zlib default,0=uncompressed,1=fast,9=best)"
+    inputBinding:
+      prefix: level=
+      separate: false
+  fasta:
+    label: fasta
+    type: int?
```
Comment (alexiswl) [inline tools/bamtofastq/2.0.0/bamtofastq__2.0.0.cwl:207]: should be boolean

File: tools/bamtofastq/2.0.0/bamtofastq__2.0.0.cwl [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+  ranges:
+    label: ranges
+    type: string?
+    doc: "input ranges (bam and cram input only, default: read complete file)"
+    inputBinding:
+      prefix: ranges=
+      separate: false
+  exclude:
+    label: exclude
+    type: string?
```
Comment (alexiswl) [inline tools/bamtofastq/2.0.0/bamtofastq__2.0.0.cwl:156]: Is this not a boolean input? Or do you specify the flag?

File: tools/bamtofastq/2.0.0/bamtofastq__2.0.0.cwl [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+  split_prefix:
+    label: split prefix
+    type: string?
+    doc: "file name prefix if collate=0 and split>0"
+    inputBinding:
+      prefix: splitprefix=
+      separate: false
+  tags:
+    label: tags
+    type: int?
```
[the remark this answers is not in the document]
Comment (skanwal) [reply tools/bamtofastq/2.0.0/bamtofastq__2.0.0.cwl:287]: may be .. not clear from help `tags=<[]> : list of aux tags to be copied (default: do not copy any aux fields)`

File: tools/bamtofastq/2.0.0/bamtofastq__2.0.0.cwl (hunk 2 above)
Comment (skanwal) [reply tools/bamtofastq/2.0.0/bamtofastq__2.0.0.cwl:156]: No - needs a flag as `exclude=<[SECONDARY,SUPPLEMENTARY]>`

File: tools/bamtofastq/2.0.0/bamtofastq__2.0.0.cwl [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+          Get the line eval bam2fastq...
+          */
+          return "eval \"bamtofastq\" '\"\$@\"'\n"
+        }
+  InitialWorkDirRequirement:
+    listing:
+      - entryname: $(get_script_path())
+        entry: |
+          #!/usr/bin/env bash
+
```
[the remark this answers is not in the document]
Comment (skanwal) [reply tools/bamtofastq/2.0.0/bamtofastq__2.0.0.cwl]: resolved in https://github.com/umccr/cwl-ica/commit/ec805dd1e00e51e95512f49c2258147036952fa8

File: tools/bamtofastq/2.0.0/bamtofastq__2.0.0.cwl [hunk 5]
```diff
@@ -0,0 +1,322 @@
...
+arguments:
+  # Script path
+  - valueFrom: "$(get_script_path())"
+    position: -1
+
+inputs:
+  first_mate:
+    label: F1
+    type: string?
+    doc: "matched pairs first mates"
```
[the remark this answers is not in the document]
Comment (skanwal) [reply tools/bamtofastq/2.0.0/bamtofastq__2.0.0.cwl]: resolved in https://github.com/umccr/cwl-ica/commit/0f99346b3c378e6a1a6f0f14d514fcd68ff74533

File: tools/bamtofastq/2.0.0/bamtofastq__2.0.0.cwl [hunk 6]
```diff
@@ -0,0 +1,322 @@
...
+  tags:
+    label: tags
+    type: int?
+    doc: "list of aux tags to be copied (default: do not copy any aux fields)"
+    inputBinding:
+      prefix: tags=
+      separate: false
+  output_per_readgroup_rgsm:
+    label: output per readgroup rgsm
+    type: int?
```
[the remark this answers is not in the document]
Comment (skanwal) [reply tools/bamtofastq/2.0.0/bamtofastq__2.0.0.cwl]: https://github.com/umccr/cwl-ica/commit/5ab9cde4e8890e6baaa5f207e5a005d8d45cec8f

File: tools/bamtofastq/2.0.0/bamtofastq__2.0.0.cwl [hunk 7]
```diff
@@ -0,0 +1,322 @@
...
+  unmatched_pairs_second_mate:
+    label: unmatched pairs mate2
+    type: string?
+    doc: "unmatched pairs second mates"
+    inputBinding:
+      prefix: O2=
+      separate: false
+  collate:
+    label: collate
+    type: int?
```
[the remark this answers is not in the document]
Comment (skanwal) [reply tools/bamtofastq/2.0.0/bamtofastq__2.0.0.cwl]: https://github.com/umccr/cwl-ica/commit/5ab9cde4e8890e6baaa5f207e5a005d8d45cec8f

File: tools/bamtofastq/2.0.0/bamtofastq__2.0.0.cwl (hunk 3 above)
[the remark this answers is not in the document]
Comment (skanwal) [reply tools/bamtofastq/2.0.0/bamtofastq__2.0.0.cwl]: https://github.com/umccr/cwl-ica/commit/5ab9cde4e8890e6baaa5f207e5a005d8d45cec8f

File: tools/bamtofastq/2.0.0/bamtofastq__2.0.0.cwl [hunk 8]
```diff
@@ -0,0 +1,322 @@
...
+      separate: false
+  output_per_readgroup_prefix:
+    label: output per read group prefix
+    type: string?
+    doc: "prefix added in front of file names if outputperreadgroup=1 (for collate=1 only)"
+    inputBinding:
+      prefix: outputperreadgroupprefix=
+      separate: false
+  output_dir:
+    label: output output_dir
```
[the remark this answers is not in the document]
Comment (skanwal) [reply tools/bamtofastq/2.0.0/bamtofastq__2.0.0.cwl]: https://github.com/umccr/cwl-ica/commit/5067a22051f61cca6be71ab43d12fe042f4b1061

File: tools/bamtofastq/2.0.0/bamtofastq__2.0.0.cwl [hunk 9]
```diff
@@ -0,0 +1,322 @@
...
+    type: string?
```
[the remark this answers is not in the document]
Comment (skanwal) [reply tools/bamtofastq/2.0.0/bamtofastq__2.0.0.cwl]: https://github.com/umccr/cwl-ica/commit/5067a22051f61cca6be71ab43d12fe042f4b1061

File: tools/bamtofastq/2.0.0/bamtofastq__2.0.0.cwl (hunk 1 above)
Comment (skanwal) [reply tools/bamtofastq/2.0.0/bamtofastq__2.0.0.cwl]: https://github.com/umccr/cwl-ica/commit/8b05df6903d1b507342b1b30cf9c0846c3c38396

File: tools/bamtofastq/2.0.0/bamtofastq__2.0.0.cwl [hunk 10]
```diff
@@ -0,0 +1,322 @@
...
+  - https://schema.org/version/latest/schemaorg-current-http.rdf
+
+# Metadata
+s:author:
+    class: s:Person
+    s:name: Sehrish Kanwal
+    s:email: sehrish.kanwal@umccr.org
+
+# ID/Docs
+id: bamtofastq--2.0.0
```
[the remark this answers is not in the document]
Comment (skanwal) [reply tools/bamtofastq/2.0.0/bamtofastq__2.0.0.cwl]: moved [rename file to reflect version update](https://github.com/umccr/cwl-ica/pull/102/commits/9c78c4af2818202eb363de4294b6be003cccfcd0)

[the remark this answers is not in the document]
Comment (alexiswl) [reply tools/bamtofastq/2.0.0/bamtofastq__2.0.0.cwl]: With the file renamed, you might need to update tool.yaml and project.yaml (either manually or re-running the cwl-ica init commands)

[the remark this answers is not in the document]
Comment (skanwal) [reply tools/bamtofastq/2.0.0/bamtofastq__2.0.0.cwl]: updated in https://github.com/umccr/cwl-ica/commit/4d729333959e5bc4c1a74538d8067a7189dc5fd0

**Review by alexiswl — APPROVED** [review]

**Pull request merged**